src/apps/competitions/tasks.py

```diff
@@ -116,17 +117,18 @@ def create_detailed_output_file(detail_name, submission):
 
 @app.task(queue='site-worker', soft_time_limit=60)
 def run_submission(submission_pk, is_scoring=False):
-    related_models = (
+    select_models = (
         'phase',
         'phase__competition',
-        'phase__input_data',
-        'phase__reference_data',
-        'phase__scoring_program',
-        'phase__ingestion_program',
-        'phase__public_data',
-        'phase__starting_kit',
     )
-    submission = Submission.objects.select_related(*related_models).prefetch_related('details').get(pk=submission_pk)
+    prefetch_models = (
+        'details',
+        'phase__tasks__input_data',
+        'phase__tasks__reference_data',
+        'phase__tasks__scoring_program',
+        'phase__tasks__ingestion_program',
+    )
+    submission = Submission.objects.select_related(*select_models).prefetch_related(*prefetch_models).get(pk=submission_pk)
```

Comment:
Hmm, I think this is OK the way it is, what do you think about this though? Will leave it up to you what style is better
```python
qs = Submission.objects.select_related(*select_models).prefetch_related(*prefetch_models)
submission = qs.get(pk=submission_pk)
```

Reply:
Yea i like your styling better. doing that.